src/modules/cpu/host_fused_functions.hpp

```diff
@@ -4,12 +4,593 @@
+        //                                                                            pTransposedCtmRow3))));
+        //         }
+
+        //         // Pack f32 back to u8
+
+        //         px0 = _mm_cvtps_epi32(pRowArr[0]);
+        //         px1 = _mm_cvtps_epi32(pRowArr[1]);
+        //         px2 = _mm_cvtps_epi32(pRowArr[2]);
+        //         px3 = _mm_cvtps_epi32(pRowArr[3]);
+
+        //         px0 = _mm_packus_epi32(px0, px1);
+        //         px1 = _mm_packus_epi32(px2, px3);
+        //         px0 = _mm_packus_epi16(px0, px1);    // pixels 0-15
+
+        //         // Shuffle and store
+
+        //         px0 = _mm_shuffle_epi8(px0, pShuffleStore); // RGBX -> RGB
+        //         _mm_storeu_si128((__m128i *)dstPtrTemp, px0);
+
+        //         srcPtrTemp += 12;
+        //         dstPtrTemp += 12;
+        //     }
+        //     for (; vectorLoopCount < bufferLength; vectorLoopCount+=3)
+        //     {
+        //         Rpp32f pixelR = ctm[0] * srcPtrTemp[0] + ctm[1] * srcPtrTemp[1] + ctm[2] * srcPtrTemp[2] + ctm[3];
+        //         Rpp32f pixelG = ctm[4] * srcPtrTemp[0] + ctm[5] * srcPtrTemp[1] + ctm[6] * srcPtrTemp[2] + ctm[7];
+        //         Rpp32f pixelB = ctm[8] * srcPtrTemp[0] + ctm[9] * srcPtrTemp[1] + ctm[10] * srcPtrTemp[2] + ctm[11];
+
+        //         *dstPtrTemp = (T) RPPPIXELCHECK(pixelR);
+        //         *(dstPtrTemp + 1) = (T) RPPPIXELCHECK(pixelG);
+        //         *(dstPtrTemp + 2) = (T) RPPPIXELCHECK(pixelB);
+
+        //         srcPtrTemp += 3;
+        //         dstPtrTemp += 3;
+        //     }
+
+        //     srcPtrTemp += increment;
+        //     dstPtrTemp += increment;
+        // }
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+        // Method 6 - WORKING - 12 elements per loop run with SSE, reduce loads with shuffle to broadcast approach - 0.001182s for average of 100 runs on 8 224x224 images
+
+        // Rpp32u bufferLength = srcSize.width * channel;
+        // Rpp32u alignedLength = bufferLength & ~11;
+
+        // __m128i const zero = _mm_setzero_si128();
+        // __m128i px0, px1, px2, px3;
+        // __m128i pShuffleLoad = _mm_setr_epi8(0, 1, 2, 12, 3, 4, 5, 13, 6, 7, 8, 14, 9, 10, 11, 15);
+        // __m128i pShuffleStore = _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14, 3, 7, 11, 15);
+
+        // __m128 pRowArr[4], pSrcF32[12];
+        // __m128 pOne = _mm_set1_ps(1.0);
+        // __m128 pTransposedCtmRow0 = _mm_load_ps(ctm); // Load ctm row 0
+        // __m128 pTransposedCtmRow1 = _mm_load_ps(ctm + 4); // Load ctm row 1
+        // __m128 pTransposedCtmRow2 = _mm_load_ps(ctm + 8); // Load ctm row 2
+        // __m128 pTransposedCtmRow3 = _mm_load_ps(ctm + 12); // Load ctm row 3
+
+        // _MM_TRANSPOSE4_PS(pTransposedCtmRow0, pTransposedCtmRow1, pTransposedCtmRow2, pTransposedCtmRow3);  // Transpose ctm to contain cols in pTransposedCtmRow0/pTransposedCtmRow1/pTransposedCtmRow2/pTransposedCtmRow3
```

Comment:
if you need the transposed matrix, write the output from previous step in that format and avoid extra compute

Reply:
Will no longer use this host_fused_functions batchPD variant